pkg/service/model_build_target_group.go

```diff
@@ -0,0 +1,312 @@
+		return intstr.FromString(rawHealthCheckPort), nil
+	}
+	var portVal int64
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCPort, &portVal, t.service.Annotations); err != nil {
+		return intstr.IntOrString{}, err
+	}
+	return intstr.FromInt(int(portVal)), nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckProtocol(_ context.Context) (elbv2model.Protocol, error) {
+	rawHealthCheckProtocol := string(t.defaultHealthCheckProtocol)
+	t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixHCProtocol, &rawHealthCheckProtocol, t.service.Annotations)
+	switch strings.ToUpper(rawHealthCheckProtocol) {
+	case string(elbv2model.ProtocolTCP):
+		return elbv2model.ProtocolTCP, nil
+	case string(elbv2model.ProtocolHTTP):
+		return elbv2model.ProtocolHTTP, nil
+	case string(elbv2model.ProtocolHTTPS):
+		return elbv2model.ProtocolHTTPS, nil
+	default:
+		return "", errors.Errorf("Unsupported health check protocol %v", rawHealthCheckProtocol)
+	}
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckPath(_ context.Context) *string {
+	healthCheckPath := t.defaultHealthCheckPath
+	t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixHCPath, &healthCheckPath, t.service.Annotations)
+	return &healthCheckPath
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckIntervalSeconds(_ context.Context) (int64, error) {
+	intervalSeconds := t.defaultHealthCheckInterval
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCInterval, &intervalSeconds, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return intervalSeconds, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckTimeoutSeconds(_ context.Context) (int64, error) {
+	timeoutSeconds := t.defaultHealthCheckTimeout
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCTimeout, &timeoutSeconds, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return timeoutSeconds, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckHealthyThresholdCount(_ context.Context) (int64, error) {
+	healthyThresholdCount := t.defaultHealthCheckHealthyThreshold
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCHealthyThreshold, &healthyThresholdCount, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return healthyThresholdCount, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckUnhealthyThresholdCount(_ context.Context) (int64, error) {
+	unhealthyThresholdCount := t.defaultHealthCheckUnhealthyThreshold
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCUnhealthyThreshold, &unhealthyThresholdCount, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return unhealthyThresholdCount, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetHealthCheck(ctx context.Context) (*elbv2model.TargetGroupHealthCheckConfig, error) {
```

Comment:
maybe `buildTargetGroupHealthCheckConfig` to align with Ingress' model builder :D

Reply:
sounds good.